drivers/w3s_test.go

```diff
@@ -0,0 +1,88 @@
+package drivers
+
+import (
+	"bytes"
+	"context"
+	"crypto/rand"
+	"fmt"
+	"github.com/google/uuid"
+	require2 "github.com/stretchr/testify/require"
+	"io"
+	"net/http"
+	"os"
+	"os/exec"
+	"testing"
+)
+
+type testFile struct {
+	dirPath string
+	name    string
+	data    []byte
+}
+
+func TestW3sOS(t *testing.T) {
+	require := require2.New(t)
+
+	ucanKey := os.Getenv("W3S_UCAN_KEY")
+	ucanProof := os.Getenv("W3S_UCAN_PROOF")
+	if ucanKey == "" || ucanProof == "" {
+		fmt.Println("No w3s credentials, test skipped")
+		return
+	}
+	_, err := exec.LookPath("w3")
+	if err != nil {
+		fmt.Println("No w3 installed, test skipped")
+		return
+	}
+	_, err = exec.LookPath("ipfs-car")
+	if err != nil {
+		fmt.Println("No ipfs-car installed, test skipped")
```

Comment:
I get that this is to stop tests breaking for people that don't have the binary, but could we do something to ensure they run on CI at least? Maybe have them on by default and an env var like `NO_W3S_TESTS=1` to not run them.

Reply:
This is always tricky for any tests related to some external system. Not sure if we should run them in CI. There are some pros and cons.

Currently, I just followed the pattern we use for all drivers, e.g., if you don't specify the `AWS_S3_SECRET` secret then it's not run against the real AWS. And I guess we don't run the AWS driver test in CI neither.

We could consider running drivers tests against real systems in our CI, but I'd prefer to do it as a separate topic,

Reply:
Fair enough, I'm okay with that